applications/dashboard/src/scripts/app/user-content/embeds/soundcloud.ts

```diff
@@ -0,0 +1,33 @@
+/**
+ * @copyright 2009-2018 Vanilla Forums Inc.
+ * @license https://opensource.org/licenses/GPL-2.0 GPL-2.0
+ */
+
+import { registerEmbed, IEmbedData, FOCUS_CLASS } from "@dashboard/embeds";
+
+// Setup soundcloud embeds.
+registerEmbed("soundcloud", soundCloudRenderer);
+
+/**
+ * Renders soundcloud embeds.
+ */
+export async function soundCloudRenderer(element: HTMLElement, data: IEmbedData) {
+    const height = data.height ? data.height : "";
+
+    const iframe = document.createElement("iframe");
+    iframe.setAttribute("id", "sc-widget");
+    iframe.setAttribute("width", "100%");
+    iframe.setAttribute("height", height.toString());
+    iframe.setAttribute("scrolling", "no");
+    iframe.setAttribute("frameborder", "no");
+    iframe.setAttribute(
+        "src",
+        data.attributes.url +
+            data.attributes.track +
```

Comment:
There are several assumptions about the presence of additional attributes here, but the data scraper isn't necessarily setting them. Track, showArtwork and visual are all conditionally set. Why not test for the presence of these attributes and use sensible defaults?